titanium-web-extension/src/executor/mod.rs

```diff
@@ -227,6 +231,41 @@ impl Executor {
         NoAction
     }
 
+    fn click_next_page(&mut self) {
+        let document = self.model.page.get_dom_document();
+        let regex = Regex::new(r"(?i:next|forward|older|more|›|»)|(?:<.+>)>(?:<.+>)").unwrap();
+
+        if let Some(document) = document {
+            if let Some(link) = match_pattern(&document, "a", regex) {
+                if let Ok(element) = link.clone().downcast::<DOMHTMLElement>() {
+                    self.click(element);
+                } else {
+                    warn!("click_next_page, Failed downcast on element");
+                }
+            } else {
+                // TODO: Check if url (not text) is *very* similar to our current one
+                // example.com/page/4 => example.com/page/5
+            }
+        }
+    }
+
+    fn click_prev_page(&mut self) {
+        let document = self.model.page.get_dom_document();
+        let regex = Regex::new(r"(?i:prev(ious)|back|newer|less|«|‹)|(?:<.+>)<(?:<.+>)").unwrap();
+
+        if let Some(document) = document {
+            if let Some(link) = match_pattern(&document, "a", regex) {
+                if let Ok(element) = link.clone().downcast::<DOMHTMLElement>() {
+                    self.click(element);
+                } else {
+                    warn!("click_next_page, Failed downcast on element");
```

Comment:
The [`wtry!` macro](https://github.com/antoyo/titanium/blob/master/titanium-web-extension/src/lib.rs#L94) already support showing an error similar to this code.